span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessorTest.java

```diff
@@ -0,0 +1,199 @@
+package org.hypertrace.core.spannormalizer.jaeger;
+
+import com.typesafe.config.ConfigFactory;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.KeyValue;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Process;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class JaegerSpanPreProcessorTest {
+
+  private final Random random = new Random();
+
+  @Test
+  void testPreProcessSpan_missingTenantId() {
+    Assertions.assertThrows(
```

Comment:
In such a case span-normalizer should not stop, it should just log and continue after dropping such messages.

Reply:
But that would be changing the existing service behaviour, I think the idea currently is to fail fast if tenant_id is not present due to missing config.